feature-libs/asm/core/store/actions/customer.action.ts

```diff
@@ -32,9 +32,12 @@ export class CustomerSearch extends StateUtils.LoaderLoadAction {
   }
 }
 
-export class CustomerSearchFail extends StateUtils.LoaderFailAction {
+export class CustomerSearchFail
+  extends StateUtils.LoaderFailAction
+  implements ErrorAction
+{
   readonly type = CUSTOMER_SEARCH_FAIL;
-  constructor(public payload: any) {
+  constructor(public error: any) {
```

Comment:
This situation is a bit tricky. We're aiming for the right data type, but there's a complication. In each Effect, errors get normalized using normalizeHttpError. This sometimes leads to the return value being undefined. To tackle this, I suggest making a new method like getError that always gives us some type of error. The main question is whether we should adjust normalizeHttpError to return the provided error instead of undefined.

Reply:
I've replaced the `any` after all. It seemed like a right thing to do.